extensions/json-language-features/client/src/jsonClient.ts

```diff
@@ -471,6 +470,22 @@ export async function startClient(context: ExtensionContext, newLanguageClient:
 		}
 	}
 
+	async function getSortTextEdits(document: TextDocument, tabSize: string | number = 4, insertSpaces: string | boolean = true): Promise<TextEdit[]> {
+		const filesConfig = workspace.getConfiguration('files', document);
+		const options: SortOptions = {
+			tabSize: Number(tabSize),
```

Comment:
Previously, this line was:
```
textEditor.options.tabSize ? Number(textEditor.options.tabSize) : 4,
```
That would ensure that if tabSize is undefined, the default would be 4. After this change, if the tabSize is undefined, the tabSize will become `NaN`

Reply:
In the parameters of the `getSortTextEdits` function the tabSize has a default value of 4. This should take into account the case when tabSize is undefined?

Reply:
I see that it can come is as undefined from this call site -> https://github.com/microsoft/vscode/blob/c2f6a1dc84055d3a8ddff0c15bc589d12b9cdc2e/extensions/json-language-features/client/src/jsonClient.ts#L172

```
const textEditsPromise = getSortTextEdits(document, documentOptions?.tabSize, documentOptions?.insertSpaces);
```

Reply:
Indeed but I assigned default values to the parameters of the function here:

```typescript
async function getSortTextEdits(document: TextDocument, tabSize: string | number = 4, insertSpaces: string | boolean = true)
````
So if tabSize is undefined, then it takes the default 4. I tested by forcing tabSize to be undefined and it takes the value 4 in the function. Perhaps there is something that I am not quite understanding.

Reply:
You're right!!